Add Code of Conduct Committee Charter

carolynvs opened the pull request:
This is a proposed charter for the Code of Conduct Committee for review by the Steering Committee.

carolynvs commented:
/hold
/sig steering

carolynvs commented:
Oops, I am awful at assigning things to teams/sigs, never get that the first time. 😊 

@michelleN Can you let the @kubernetes/steering-committee know that this is ready to review again?

nikhita commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+
+Steering committee members may also act in limited cases to enforce the code of
+conduct. Examples of this may be the deletion of GitHub comments, offensive
+Slack messages, or the eviction of bad actors from public meetings. These
```
One small point here - the steering committee might not have powers to delete GitHub comments in some repos (github admin team can do that) or delete slack messages (slack admin team can do that).

timothysc replied:
Most do, but we should double check that it exists across all k8s-orgs.  You may want to file an issue https://github.com/kubernetes/org

spiffxp replied:
I do not agree that steering committee members should implicitly have Owner or even Admin rights across all of our orgs and all of our repos.  Should steering have the ability to ask the appropriate teams to take action? Sure.  But at least for all k8s orgs, we delegated the responsibilities to the GitHub Admin team, we should not subvert that.  We addressed this in https://github.com/kubernetes/org/issues/637

timothysc commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+# Kubernetes Code of Conduct Committee Charter
+
+## Mission/Purpose
+Our primary concern is creating and maintaining a safe and respectful community.
```
s/concern/mission

eparis commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+If committee members believe that the committee is no longer able to act in
+accordance with the above Mission/Purpose the committee may vote to dissolve.
+The committee should specify a date of dissolution. Dissolution requires an
+affirmative vote of more than 75% of committee members. If the committee is
```
Unanimous was my intention if there were only 4 members.

carolynvs replied:
Is this resolved or does the charter need the wording clarified?

spiffxp replied:
Personally I would be explicit that it requires unanimity if the committee drops below 5 members

carolynvs replied:
I've added wording to cover the case for < 5 members.

carolynvs commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+members](https://git.k8s.io/community/committee-code-of-conduct) elected by the
+Steering Committee based on private nomination. During an election, the two
+candidates that received the most committee votes are granted 2 year terms, with
+the remaining person given a one year term. 
```
I don't believe that what you listed at the top is correct. Here's how I understand our term rotations and elections, you can see it already spelled out in our readme based on our own term limits: https://github.com/kubernetes/community/tree/master/committee-code-of-conduct#terms. Jaice and Paris both have 2 year terms, and myself, Jennifer and Eric have 1 year terms that end this summer.

The top two winners of the election get 2 year terms, the remaining people get 1 year terms. I believe that works out to this?

* year 1: A,B,C,D,E (A,B are 2 year terms)
* year 2: A,B,F,G,H (F,G are 2 year terms)
* year 3: F,G,I,J,K (I,J are 2 year terms)
* year 4: I,J,L,M,N (L,M are 2 year terms)

The first year of the committee is the exception where multiple people received a 1 year term, but going forward, we always elect enough people to replace the departing people, so it's 3 people elected, two of them getting 2year terms, and one getting a 1year term. Giving us overlap between old and new members to preserve knowledge and continuity.

Does that help?

spiffxp replied:
Yes, your summary paragraph at the end is a clearer description of the intended process than what's written in the charter.

carolynvs replied:
I've updated the charter to attempt to clarify this with some of my wording above.

carolynvs commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+Meetings are considered at quorum when a simple majority of the members are
+present. Where there are 4 or fewer members available due vacant seats or
+recusal, quorum is 2. Note that many decisions are not able to be implemented in
+situations where only 2 members are present.
```
@eparis I think that sentence used to make sense in an older draft. But after we trimmed down parts of the document, it isn't necessary anymore. Do you agree?

carolynvs replied:
I've removed the old wording since it's not needed anymore.

carolynvs commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,150 @@
+situations where only 2 members are present.
+
+### Policy change ratification
+Any changes to the charter require explicit LGTM or Approve from all active
```
In my opinion, we don't need that word there. I believe it's a leftover from an older draft.

carolynvs replied:
I've removed the word active.

nikhita commented:
This needs another rebase :(

carolynvs commented:
Thanks for the nudge @nikhita! I've rebased and addressed all the feedback that I could.

philips commented:
This LGTM. Although I haven't been following along as closely as @spiffxp

spiffxp commented:
/lgtm

philips commented:
I am going to unhold this after the Wednesday steering committee if no one has any other comments.

dims commented:
Sounds good @philips !

spiffxp commented:
https://github.com/kubernetes/community/issues/3353 is about spelling out an appointment/election process.  This charter spells out terms, but very little details of an actual election process, such as:
- when are candidates chosen?
- how are candidates chosen?
- when does voting occur?
- how does voting occur? (I believe we used CIVS last time)

The linked issue mentions the idea of nominating candidates this month (June), a vote on candidates before the end of July, and a transition by August 7th.  If you're comfortable with process described there, can we flesh out the charter to include these details? @carolynvs @jdumars

timothysc commented:
> This charter spells out terms, but very little details of an actual election process, such as: 

IMO, follow on with a separate issue+doc.  
Elections should typically not be part of the charter and are subject to change.

carolynvs commented:
I have updated the charter to remove the election process and moved it into a separate document (which will be filled in later, hopefully very soon!)

spiffxp commented:
/lgtm
August 7th is less than a month away, I question whether we're going to make that deadline.

deads2k commented on committee-code-of-conduct/charter.md:
```diff
@@ -0,0 +1,149 @@
+policy document and is supported with additional references and tools as needed.
+Since maintaining a safe environment is a very large part of what the committee
+does, we must carefully balance transparency of process with preserving the
+privacy of those who come to the committee or other community leaders with an
```
I think the code of conduct committee should also consider the privacy of those accused of causing the incident.

eparis replied:
We've discussed and agree this language should be more broad. We do say, further down, that we intended the privacy to be for everyone, including the accused.

We will send a followup PR (after this merges) to discuss and address this specific area.

timothysc commented:
Canceling hold per timeout above. 
PRs can, and should, be made to this initial draft.  Other comments look like they can be addressed in that fashion. 
 
/hold cancel 
/cc @michelleN
